source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -110,6 +110,30 @@ namespace PlayFab
             auto playFabEmitEventResponse = std::make_shared<PlayFabEmitEventResponse>();
             playFabEmitEventResponse->emitEventResult = emitResult;
 
+            std::shared_ptr<PlayFabError> emitEventError  = std::make_shared<PlayFabError>();
+
+            emitEventError->ErrorName = "PlayFabEventPipeline IntakeEvent Error";
+            emitEventError->ErrorMessage = "PlayFabEventPipeline did not accept the event. Please see ErrorDetails for more information.";
+            emitEventError->ErrorCode = PlayFabErrorCode::PlayFabErrorEventNotFound;
+
+            emitEventError->HttpCode = 0;
+            emitEventError->HttpStatus = "None";
+
+            if(emitResult == EmitEventResult::Overflow)
```

Comment:
>if [](start = 12, length = 2)

nit: instead of adding a new sequence of if-else statements, why not use the switch case above which sets `emitResult`

Reply:
That will require moving the creation of the PlayFabError to before the previous switch statement. I originally thought that would look nicer too, but I think there are going to be more non error causing cases of this IntakeEvent call, which an empty event construction would be unnecessary for the most common expected case.

Reply:
I don't think it would. The only thing that needs to be declared before the switch is a variable to hold the string literal.
e.g.
```
const char* errorDetail = "";
switch (this->buffer.TryPut(request))
{
... // set errorDetail in switch
}

auto emitEventError = std::make_share<>()
// ...
emitEventError->ErrorDetails = errorDetail;
```

that being said this is all personal preference so i'll leave it to your discretion